sound/soc/intel/boards/sof_ssp_amp.c

```diff
@@ -239,25 +166,24 @@ static int sof_ssp_amp_probe(struct platform_device *pdev)
 
 		if (mach->mach_params.codec_mask & IDISP_CODEC_MASK)
 			ctx->hdmi.idisp_codec = true;
-
-		sof_ssp_amp_card.num_links += ctx->hdmi_num;
 	} else {
 		ctx->hdmi_num = 0;
 	}
 
-	if (sof_ssp_amp_quirk & SOF_SSP_BT_OFFLOAD_PRESENT) {
+	if (sof_ssp_amp_quirk & SOF_SSP_BT_OFFLOAD_PRESENT)
 		ctx->bt_offload_present = true;
-		sof_ssp_amp_card.num_links++;
-	}
 
-	dai_links = sof_card_dai_links_create(&pdev->dev, ctx->amp_type,
-					      ctx->ssp_amp, ctx->dmic_be_num,
-					      ctx->hdmi_num,
-					      ctx->hdmi.idisp_codec);
-	if (!dai_links)
-		return -ENOMEM;
+	ctx->link_order_overwrite = NO_CODEC_LINK_ORDER;
 
-	sof_ssp_amp_card.dai_link = dai_links;
+	if (ctx->ssp_mask_hdmi_in) {
+		/* the topology supports HDMI-IN uses fixed BE ID for DAI links */
+		ctx->link_id_overwrite = NO_CODEC_LINK_IDS;
```

Comment:
Hmm, what if `ctx->ssp_mask_hdmi_in` is true, but `ctx->amp_type` != `CODEC_NONE`? In that case, the second be_id will still be `SPK_BE_ID`, right?

Reply:
In this case, amp's BE ID will always be SPK_BE_ID regardless the number of HDMI-IN links.